docs/guide/build-life.md

```diff
@@ -23,4 +23,12 @@ Other chores such as health checking of the agent and GitHub status reporting ha
 
 Straight forward and nothing magical. :sweat_smile:
 
+## Visual Representation
+
+Too many words? Here is a diagram as to how builds are executed:
+
+![diagram](https://raw.githubusercontent.com/RDIL/cirrus-ci-diagrams/master/cirrus-flow-diagram.jpg)
```

Comment:
The flow is a bit more complicated then the diagram and does't always go that way because of possibly custom execution behaviour. I'd suggest to remove it for now and if you want iterate on the diagram in a separate PR.

Reply:
Fixed.

Reply:
Maybe I'll remake the diagram later, but for now I removed it, so this should be ready.